oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/data/ReadWriteSafeCache.java

```diff
@@ -34,9 +34,13 @@
      */
     private volatile BufferedData<T> writeBufferPointer;
     /**
-     * Read/Write lock.
+     * Write-buffer lock.
+     *
+     * Access of read-buffer is no need to be locked,
+     * because only {@link org.apache.skywalking.oap.server.core.storage.PersistenceTimer}
+     * thread reads data from read-buffer. No concurrency reading exists.
      */
-    private final ReentrantLock lock;
+    private final ReentrantLock writeLock;
```

Comment:
Why rename this field? Basically, we don't change anything related to this.

Reply:
The lock is still being used in read and write, please revert the name.

Reply:
Sorry, I'll revert it.